From: Lorenzo Stoakes <lstoakes@gmail.com>
Subject: mm-vmalloc-dont-purge-usable-blocks-unnecessarily-fix
Date: Mon, 5 Jun 2023 21:44:20 +0100

We do NOT purge if this conditional evalutates true, so correct
VMAP_PURGE_THRESHOLD check.

Link: https://lkml.kernel.org/r/3e92ef61-b910-4576-88e7-cf43211fd4e7@lucifer.local
Signed-off-by: Lorenzo Stoakes <lstoakes@gmail.com>
Cc: Baoquan He <bhe@redhat.com>
Cc: Christoph Hellwig <hch@lst.de>
Cc: Thomas Gleixner <tglx@linutronix.de>
Cc: Uladzislau Rezki (Sony) <urezki@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vmalloc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/vmalloc.c~mm-vmalloc-dont-purge-usable-blocks-unnecessarily-fix
+++ a/mm/vmalloc.c
@@ -2102,7 +2102,7 @@ static bool purge_fragmented_block(struc
 		return false;
 
 	/* Don't overeagerly purge usable blocks unless requested */
-	if (!force_purge && vb->free < VMAP_PURGE_THRESHOLD)
+	if (!(force_purge || vb->free < VMAP_PURGE_THRESHOLD))
 		return false;
 
 	/* prevent further allocs after releasing lock */
_
